File: frontend/src/types/itinerary.ts
export interface Activity {
  name: string;
  category: string;
  lat: number;
  lng: number;
  estimated_duration_hrs: number;
  cost_estimate_usd: number;
  opening_hours: string;
  rating: number;
  description: string;
  start_time: string;
  end_time: string;
}

export interface DayPlan {
  day_number: number;
  date: string;
  activities: Activity[];
  travel_times_min: number[];
  weather_summary: string;
  total_cost_usd: number;
}

export interface Itinerary {
  _id: string;
  trip_id: string;
  version: number;
  destination_info: string;
  currency: { code: string; rate_to_usd: number };
  weather_forecast: WeatherDay[];
  days: DayPlan[];
  total_cost_usd: number;
}

export interface WeatherDay {
  date: string;
  temp_max: number;
  temp_min: number;
  precipitation: number;
  weathercode: number;
}

File: backend/app/models/itinerary.py
from typing import Optional

from pydantic import BaseModel, Field


class Activity(BaseModel):
    name: str
    category: str
    lat: float
    lng: float
    estimated_duration_hrs: float
    cost_estimate_usd: float
    opening_hours: str = ""
    rating: float = 0.0
    description: str = ""
    start_time: str = ""
    end_time: str = ""


class DayPlan(BaseModel):
    day_number: int
    date: str
    activities: list[Activity]
    travel_times_min: list[int] = []
    weather_summary: str = ""
    total_cost_usd: float = 0.0


class ItineraryResponse(BaseModel):
    id: str = Field(alias="_id")
    trip_id: str
    version: int = 1
    destination_info: str = ""
    currency: dict = {}
    weather_forecast: list[dict] = []
    attractions: list[dict] = []
    days: list[DayPlan] = []
    total_cost_usd: float = 0.0

    class Config:
        populate_by_name = True
